### PR TITLE
Add GetAttesterSlashingV2

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
This PR introduces the skeleton of the new method GetAttesterSlashingV2. Note though that as of now this differ minimally from the V1 as Teku currently doesn't not make a distinction of Electra.IndexedAttestation and Phase0.IndexedAttestation since the only difference is the size limit of the `attesting_indices` array. 
This implementation does not make us 100% complaint with the spec but should result in the right json result after all.

## Fixed Issue(s)

Part of 
 
## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v2/beacon/GetAttesterSlashingV2Test.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  @Test
+  void metadata_shouldHandle400() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_BAD_REQUEST);
+  }
+
+  @Test
+  void metadata_shouldHandle500() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_INTERNAL_SERVER_ERROR);
+  }
```
Comment (rolfyone) [inline data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v2/beacon/GetAttesterSlashingV2Test.java]: should be able to test handling 200 even without implementation data :)

File: data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/schema/GetPoolAttesterSlashingsV2Response.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      ]
+    },
+    "data": {
+      "type": "object",
+      "oneOf": [
+        {
+          "type": "array",
+          "items": {
+            "$ref": "#/components/schemas/AttesterSlashing"
+          }
```
Comment (rolfyone) [inline data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/schema/GetPoolAttesterSlashingsV2Response.json:24]: see if we can get attester slashing for electra to work here too...

also no mention of the `Eth-Consensus-Version` header in the response...

Comment (gfukushima) [reply data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/schema/GetPoolAttesterSlashingsV2Response.json:24]: Eth-Consensus-Version will be added once the implementation is working. It's prob going to come down to pulling the milestone from one of the AttesterSlashing data.

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v2/beacon/GetAttesterSlashingV2Test.java (hunk 1 above)
Comment (gfukushima) [reply data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v2/beacon/GetAttesterSlashingV2Test.java]: added a 200 for phase 0 for now. Will add electra in the next PR

File: data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/schema/GetPoolAttesterSlashingsV2Response.json (hunk 2 above)
Comment (rolfyone) [reply data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/schema/GetPoolAttesterSlashingsV2Response.json:24]: its visible in metadata, we should just add it to the definition so it comes up...

Comment (rolfyone) [reply data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/schema/GetPoolAttesterSlashingsV2Response.json:24]: actually is it automatic... i cant find it in any gets... hrm...

**Review by rolfyone — APPROVED** [review]

**Pull request merged**